python/golem_task_api/client.py

```diff
@@ -184,16 +185,18 @@ def __init__(
             task_id: str,
             max_subtasks_count: int,
             task_params: dict,
-    ) -> Task:
+    ) -> Tuple[Task, Infrastructure]:
```

Comment:
Returning a protobuf message from client method breaks the principle of hiding GRPC protocol details. There should be an `Infrastructure` struct same as `Task`.
Additionally, we can consider putting infrastructure requirements as a field of `Task` struct. Why return a tuple when we have a tailored return type for this particular method.

Reply:
> There should be an Infrastructure struct same as Task.

This new `Infrastructure` class will be a duplicate of the `Infrastructure` message. We can maintain both (as I've implemented initially).

> Returning a protobuf message from client method breaks the principle of hiding GRPC protocol details.

What protocol details does it expose?

> Why return a tuple when we have a tailored return type for this particular method.

To not include minimum hardware requirements that are orthogonal to task environment definition.